w3ctrace/state.go

```diff
@@ -19,13 +19,34 @@ type State []string
 func ParseState(traceStateValue string) (State, error) {
 	var state State
 
-	for _, st := range strings.SplitN(traceStateValue, ",", 32) {
-		st = strings.TrimSpace(st)
-		if st == "" {
-			continue
+	// comasLimit is used to prevent handling of malformed values that contain a lot of comas
+	const comasLimit = 256
+	rawEntries := strings.SplitN(traceStateValue, ",", comasLimit)
+
+	rawEntriesLen := len(rawEntries)
+	if rawEntriesLen > 32 {
+		filtered := 0
+		for _, st := range rawEntries {
+			if rawEntriesLen-filtered <= 32 || len(state) == 32 {
+				break
+			}
+
+			st = strings.TrimSpace(st)
+			if len(st) > 128 || st == "" {
+				filtered++
+				continue
+			}
+
+			state = append(state, st)
+		}
+	} else {
+		for _, st := range rawEntries {
+			st = strings.TrimSpace(st)
```

Comment:
Same as above, this would alter a value that we don't own.

Reply:
Answered in the previous comment.